docs/reference/tools-reference/open_source_llm_tool.md

```diff
@@ -0,0 +1,59 @@
+# Open Source LLM
+
+## Introduction
+
+The Prompt flow Open Source LLM tool enables you to utilize a variety of Open Source and Foundational Models, such as [Falcon](https://aka.ms/AAlc25c) or [Llama 2](https://aka.ms/AAlc258) for natural language processing, in PromptFlow.
+
+This Prompt flow supports two different LLM API types:
+
+- **Completion**: to generate text based on provided prompts.
+- **Chat**: to facilitate interactive conversations with text-based inputs and responses.
+
+## Prerequisite
+
+1. Pick the model to use with from the [Azure Machine Learning model catalog](https://ml.azure.com/model/catalog).
+2. Use the "Deploy" button to deploy the model to AzureML Inference endpoint.
+
+More detailed instructions can be found here [Deploying foundation models to endpoints for inferencing.](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-use-foundation-models?view=azureml-api-2#deploying-foundation-models-to-endpoints-for-inferencing)
+
+## **Connections**
+
+Setup connections to provisioned resources in prompt flow.
+
+| Type        | Name     | API KEY  | API Type | API Version |
+|-------------|----------|----------|----------|-------------|
+| CustomConnection | Required | Required | -        | -           |
+
+Instructions to create a Custom Connection [can be found here.](https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/how-to-integrate-with-langchain?view=azureml-api-2#create-a-connection)
```

Comment:
This is a doc about Integrating with LangChain, and it does not have VS Code Extension creation method. [This doc](https://microsoft.github.io/promptflow/how-to-guides/manage-connections.html) might be more suitable.

Reply:
It's a value to determine which type of model the tool is calling which we want the user to set as part of the connection.
I've added more details here.